llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1778,10 +1778,9 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
       break;
     }
 
-    if (IndexOp.getImm() != 0 &&
-        SrcTy.getElementCount().getKnownMinValue() % IndexOp.getImm() != 0) {
-      report("Index must be a multiple of the source vector's minimum vector "
-             "length",
+    if (IndexOp.getImm() % DstTy.getElementCount().getKnownMinValue() != 0) {
```

Comment:
Please add the equivalent of this from the IR verifier

```
    // If this extraction is not the 'mixed' case where a fixed vector is        
    // extracted from a scalable vector, ensure that the extraction does not     
    // overrun the parent vector.                                                
    if (VecEC.isScalable() == ResultEC.isScalable()) {                           
      Check(IdxN < VecEC.getKnownMinValue() &&                                   
                IdxN + ResultEC.getKnownMinValue() <= VecEC.getKnownMinValue(),  
            "vector_extract would overrun.");                                    
    }  
```

Reply:
`mixed` is currently not documented/supported in GlobalIsel.

Reply:
Then the verifier needs to check that the scalable bits match.